capnproto-codec.sc

```diff
@@ -106,13 +106,13 @@ enum Pointer
     fn to-word (self)
```

Comment:
it occurs to me that this whole function isn't handling offsets correctly; The serialized pointers should have a signed thirty bit offset from where the pointer lives in the buffer to the data location. Well, the near pointers at least. This is trying to put the absolute position of the referenced data into the word.

Reply:
because for convenience I made the decision that when we extract a pointer from a buffer, the location of the pointer in memory and the offset the pointer points to should be combined into a single value. So when turning it back into a word it needs to know where the word is going to go to recreate the offset from the pointer to the target.